src/ResourceGenerator/RouteBasedCollectionStrategy.php

```diff
@@ -100,7 +100,7 @@ protected function generateSelfLink(
         Metadata\AbstractCollectionMetadata $metadata,
         ResourceGenerator $resourceGenerator,
         ServerRequestInterface $request
-    ) {
+    ) : Link {
```

Comment:
Must be removed because the class and / or the method are not final and can be overwritten in userland code.

Reply:
I can remove it, but it will also need to be removed from the `src/ResourceGenerator/ExtractCollectionTrait.php` since it fails to run in PHP 8 due to the parent method being different:

```sh
PHP Fatal error:  Declaration of Mezzio\Hal\ResourceGenerator\RouteBasedCollectionStrategy::generateSelfLink(Mezzio\Hal\Metadata\AbstractCollectionMetadata $metadata, Mezzio\Hal\ResourceGenerator $resourceGenerator, Psr\Http\Message\ServerRequestInterface $request) must be compatible with Mezzio\Hal\ResourceGenerator\ExtractCollectionTrait::generateSelfLink(Mezzio\Hal\Metadata\AbstractCollectionMetadata $metadata, Mezzio\Hal\ResourceGenerator $resourceGenerator, Psr\Http\Message\ServerRequestInterface $request): Mezzio\Hal\Link in /opt/project/src/ResourceGenerator/RouteBasedCollectionStrategy.php on line 99

```

Reply:
Well, removing a type-hint is a BC break and thus needs a major version bump.

Is it possible to throw the unused method "away" by using this work around:

```php
use ExtractCollectionTrait {
    generateSelfLink as unusedGenerateSelfLink;
}
```

This method is 100% overridden by the implementing class and thus should work.

I was not even aware that methods from traits are not imported when a method already exists, I've thought one has to explicitly use the `use` statement to rename a duplicated method from within a trait...

PHP always gives me bad feeling...